static/js/src/bg-form-submit.js

```diff
@@ -0,0 +1,205 @@
+    }
+  }
+
+  // Add a thank-you fallback notification to the page
+  // if someone submitted a form without a thank you action
+  if (return_url === null && isModal === false && isWhitepaper === false) {
+    if (thankYouMessage === null) {
+      thankYouMessage =
+        "Thank you<br />A member of our team will be in touch within one working day";
+    }
+    const feedbackArea = document.querySelector(".js-feedback-notification");
+    if (feedbackArea) {
+      feedbackArea.innerHTML =
+        '<div class="p-notification--positive"><p class="p-notification__response">' +
+        thankYouMessage +
+        "</p></div>";
+      const inputs = marketoForm.querySelectorAll("input, button");
+      for (var i = 0; i < inputs.length; i++) {
+        inputs[i].disabled = "disabled";
+      }
+      const recaptcha = marketoForm.querySelector(".g-recaptcha");
+      if (recaptcha) {
+        recaptcha.classList.add("u-hide");
+      }
+      marketoForm.style.opacity = ".5";
+    } else {
+      document
+        .getElementById("main-content")
+        .insertAdjacentHTML(
+          "afterbegin",
+          '<div class="p-strip is-shallow u-no-padding--bottom"><div class="row"><div class="p-notification--positive"><p class="p-notification__response">' +
+            thankYouMessage +
+            "</p></div></div></div>"
+        );
+      window.scrollTo(0, 0);
+    }
+  }
+
+  if (isWhitepaper) {
+    whitepaperAfterSubmit();
+  }
+};
+
+const backgroundSubmitHandlerClosure = function () {
+  return function (submitEvent) {
+    // Prevent normal submit
+    submitEvent.preventDefault
+      ? submitEvent.preventDefault()
+      : (submitEvent.returnValue = false);
+
+    // get form
+    const marketoForm = document.getElementById(submitEvent.target.id);
+
+    // Change the form's action location
+    marketoForm.action =
+      "https://app-sjg.marketo.com/index.php/leadCapture/save2";
+
+    // Submit the form in the background
+    backgroundSubmit(marketoForm);
+  };
+};
+
+const backgroundSubmit = function (marketoForm, submitCallback) {
+  const request = new XMLHttpRequest();
+  const submitUrl = marketoForm.getAttribute("action");
+  let formData = serialize(marketoForm);
+
+  request.open("POST", submitUrl, true);
+
+  // Send the proper header information along with the request
+  request.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
+
+  // When request has finished, call the callback function
+  if (submitCallback) {
+    request.addEventListener("readystatechange", function () {
+      if (this.readyState == 4) {
+        // Pass context and arguments on to submitCallback
+        submitCallback.apply(this, arguments);
+      }
+    });
+  }
+
+  // recaptcha test
+  const recaptchaListElement = marketoForm.getElementsByClassName(
+    "g-recaptcha"
+  )[0];
+  const recaptchaWidgetId = recaptchaListElement.dataset.widgetId;
+  const response = grecaptcha.getResponse(recaptchaWidgetId);
```

Comment:
I don't see how there could be a race condition on this as this function is only called when the form is submitted. I think we should leave the recapture loading per form and dynamically for interactive forms.

Reply:
Ok, reverted.